src/Core/Grid/Definition/Factory/MerchandiseReturnProductsGridDefinitionFactory.php

```diff
@@ -0,0 +1,255 @@
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MerchandiseReturnProductException;
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\BulkActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\Type\SubmitBulkAction;
+use PrestaShop\PrestaShop\Core\Grid\Action\ModalOptions;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\SubmitRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\MerchandiseReturnCustomizationColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class MerchandiseReturnGridDefinitionFactory builds grid definition for merchandise returns grid.
+ */
+final class MerchandiseReturnProductsGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    const GRID_ID = 'merchandise_return_products';
+
+    /**
+     * @var int
+     */
+    private $merchandiseReturnId;
+
+    /**
+     * MerchandiseReturnProductsGridDefinitionFactory constructor.
+     *
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param RequestStack $requestStack
+     */
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher,
+        RequestStack $requestStack
+    ) {
+        parent::__construct($hookDispatcher);
+        $this->setMerchandiseReturnId($requestStack);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Merchandise return products', [], 'Admin.Orderscustomers.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('merchandise_return_bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_order_detail',
+                        'extra_fields' => [
+                            'id_customization',
+                        ],
+                    ])
+            )
+            ->add(
+                (new DataColumn('product_reference'))
+                    ->setName($this->trans('Reference', [], 'Admin.Global'))
```

Comment:
Hi @JevgenijVisockij, could I have a screenshot of where this text would appear please? Thank you. :)

Reply:
![image](https://user-images.githubusercontent.com/47781757/86892114-7f6d7380-c108-11ea-99b1-cd521470b335.png)
Hey @Junebyun  and @LouiseBonnard you can see Reference and Customizations in the screenshot. Those are headers of the list in the bottom part of the screenshot.